File: CS50x/pset6/readability/readability.py
# Inputting text
text = input("Text: ")

# Initialising count of number of letters, words, and sentences to be 0
l = w = s = 0

# Iterating through text character by character
for i in text:
    # If character is a alphabet, increment count of letters by 1
    if i.isalpha():
        l = l + 1
    # If character is a space, increment count of words by 1
    elif i == " ":
        w = w + 1
    # If character is a period or exclamation or question mark, increment count of sentences by 1
    elif i in [".", "!", "?"]:
        s = s + 1
# Increment once for the first word of text
w = w + 1

# L is the average number of letters per 100 words in the text
L = (l / w) * 100
# S is the average number of sentences per 100 words in the text.
S = (s / w) * 100

# Coleman-Liau index formula
index = round(0.0588 * L - 0.296 * S - 15.8)

# Deciding grade-level for reading and printing it
if (index >= 16):
    print("Grade 16+")
elif (index > 0):
    print("Grade", index)
else:
    print("Before Grade 1")

File: CS50x/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main()
{
    // Taking a text as input
    string text = get_string("Text: ");
    int letters = 0, words = 1, sentences = 0, index;
    float L, S;

    // Calculating letters, words and sentences
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // Calculating required parameters
    L = ((float)letters / (float)words) * 100;
    S = ((float)sentences / (float)words) * 100;

    // Using Coleman-Liau index formula
    index = round(0.0588 * L - 0.296 * S - 15.8);

    // Printing according to value of index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }

}